src/openassetio-core/include/openassetio/trait/BlobTrait.hpp

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2022 The Foundry Visionmongers Ltd
+/**
+ * Define the core `BlobTrait` trait class.
+ */
+#pragma once
+
+#include <openassetio/export.h>
+
+#include "TraitBase.hpp"
+
+#include "property.hpp"
+
+namespace openassetio {
+inline namespace OPENASSETIO_VERSION {
+namespace trait {
+
+/**
+ * Core trait class representing a locatable blob of data of a specific
+ * type.
+ *
+ * Has the ID of "blob" and defines "url" and "mimeType" properties.
+ */
+struct OPENASSETIO_CORE_EXPORT BlobTrait : TraitBase<BlobTrait> {
+  /// ID of this trait.
+  static inline const TraitId kId{"blob"};
+  /// Key for URL property.
+  static inline const property::Key kUrl{"url"};
```

Comment:
[consider] Should we prefix property key names to avoid ambiguity? eg `kProperty_url` or similar, so that it is clear that `kId` isn't a property called `id`?

Reply:
I went a slightly different direction in 49c7861c, where I made `kUrl` and `kMimeType` private implementation details.  My argument is that if we do keep them as public members, then we should also expose them to Python.  So, since YAGNI I went with just making them private.

Reply:
Should these be private to the implementation of `BlobTrait`, not to the API? _Any_ trait must be considered out-of-core so we can't have anything shared/etc.. that can't be done in a totally random, unconnected repo with just a copy of the source.

Reply:
Also, was the removal of these to force the use of a traits as a view on a `Specification` instance, instead of using their key constants directly in `Specification::getTraitProperty`? Though higher overhead, that's probably a 'good thing' as there have been comments before that more than one way to do the same thing causes confusion.

Reply:
The only reason they were public was so the Python bindings could use them in crafting an exception message, which seemed like a bad excuse for making something public and, like you say, unintentionally encouraging their use as keys in a raw `getTraitProperty` query.

Good point that we want to keep the traits self-contained, though.  I can rename the `keys.hpp` to `BlobTrait.hpp` so it's clear.

Reply:
Updated in 40cf7353

Reply:
... and then removed in 077c393f after off-line discussion, where we decided that simplifying the exceptions is a small price to pay for dodging this question entirely ;)